go/jsonclient/client.go

```diff
@@ -0,0 +1,191 @@
+package jsonclient
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	ct "github.com/google/certificate-transparency/go"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// JSONClient provides common functionality for interacting with a JSON server
```

Comment:
JSONClient seems a slightly odd name.  It's still an HTTP client but one that expects requests and responses to be JSON documents.  Can't think of a better name at the moment.

Maybe the reason this feels a bit odd is that there are two responsibilities being taken by this struct: being an HTTP client with some extra behaviors like back-off, and being a thing that sends and receives JSON.  Might these responsibilities be separated?